### PR TITLE
Redid builtin_cd to return int

### PR DESCRIPTION
**Pull request by sheepman39**

### CODE REVIEW

File: src/impl.cc [hunk 1]
```diff
@@ -23,7 +23,7 @@ std::string current_line;
 struct DictStruct
 {
     std::string keyword;
-    void (*functionPointer)(int argc, char *argv[]);
+    int (*function_pointer)(int argc, std::string *argv);
```
Comment (dennorak) [inline src/impl.cc:26]: This should be `int argc, char** argv`

File: src/impl.cc [hunk 2]
```diff
@@ -283,7 +283,7 @@ std::string _get_current()
 
 // CD COMMANDS
 
-std::string builtin_cd(int argc, std::string *argv)
+int builtin_cd(int argc, std::string *argv)
```
Comment (dennorak) [inline src/impl.cc:286]: same as the other comment

**Review by dennorak — CHANGES_REQUESTED** [review]
See comments, but basically let's pass a `char**` instead of `std::string`

**Comment by sheepman39** [comment]
Should be good to merge now.

**Pull request merged**